### PR TITLE
Contrib/mvn

### PR DESCRIPTION
**Pull request by uguy**
Proposed improvement on maven aspects

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -92,70 +108,119 @@
...
-                        <phase>package</phase>
+                        <phase>generate-resources</phase>
                         <goals>
-                            <goal>single</goal>
+                            <goal>execute</goal>
                         </goals>
                         <configuration>
-                            <descriptor>${basedir}/assembly.xml</descriptor>
-                            <appendAssemblyId>false</appendAssemblyId>
+                            <source>${project.basedir}/src/main/build/dependencies-as-var.groovy</source>
```
Comment (alachambre) [inline pom.xml]: I can't find this groovy script?

Comment (uguy) [reply pom.xml]: Oups, build folder are ignored. I rename it and let you see :)

File: pom.xml [hunk 2]
```diff
@@ -92,70 +108,119 @@
...
+                    <configuration>
+                        <failOnNoGitDirectory>false</failOnNoGitDirectory>
+                        <failOnUnableToExtractRepoInfo>false</failOnUnableToExtractRepoInfo>
+                        <generateGitPropertiesFile>true</generateGitPropertiesFile>
+                        <commitIdGenerationMode>full</commitIdGenerationMode>
+                    </configuration>
+                </plugin>
+                <plugin>
+                    <groupId>org.codehaus.gmaven</groupId>
+                    <artifactId>groovy-maven-plugin</artifactId>
```
Comment (rbioteau) [inline pom.xml:184]: Why using another groovy compiler ?
I think this one doesn't work as well in eclipse ide

File: pom.xml [hunk 3]
```diff
@@ -2,19 +2,26 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
-    <parent>
```
Comment (rbioteau) [inline pom.xml:6]: 👍

File: pom.xml [hunk 4]
```diff
@@ -29,6 +36,15 @@
         <spock-core.version>1.3-groovy-2.4</spock-core.version>
         <logback-classic.version>1.2.3</logback-classic.version>
 
+        <!-- Maven plugins -->
+        <maven-compiler-plugin.version>3.8.1</maven-compiler-plugin.version>
+        <maven.compiler.source>${java.version}</maven.compiler.source>
+        <maven.compiler.target>${java.version}</maven.compiler.target>
+        <maven-assembly-plugin.version>3.1.1</maven-assembly-plugin.version>
+        <groovy-maven-plugin.version>2.1.1</groovy-maven-plugin.version>
```
Comment (rbioteau) [inline pom.xml:45]: Either use groovy maven plugin or groovy eclipse compiler

File: pom.xml [hunk 5]
```diff
@@ -92,70 +108,119 @@
...
+                            </configuration>
+                        </execution>
+                    </executions>
+                </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>${maven-compiler-plugin.version}</version>
+                    <configuration>
+                        <compilerId>groovy-eclipse-compiler</compilerId>
```
Comment (rbioteau) [inline pom.xml:149]: Either use groovy maven plugin or groovy eclipse compiler

**Review by rbioteau — COMMENTED** [review]
I like the idea of getting rid of the parent pom dependency
The usage of the git commit id plugin seems to specific to be included in the archetype imho

File: pom.xml (hunk 5 above)
Comment (uguy) [reply pom.xml:149]: The groovy maven plugin is not used to compile code but to generate the dependency list in xml format

File: pom.xml (hunk 2 above)
Comment (uguy) [reply pom.xml:184]: ok for me to not include  git commit id plugin but I would like to have the default gitignore and gitattribute already in place

**Pull request merged**